### PR TITLE
Better Physics

### PR DESCRIPTION
**Pull request by iiAzui**

### CODE REVIEW

File: Sprites/Copy of SHH Teacher Rec_Ackwledgement form - Google Docs.pdf
Comment (lishaduck) [inline Sprites/Copy of SHH Teacher Rec_Ackwledgement form - Google Docs.pdf:1]: Good catch!

File: wizard.gd [hunk 1]
```diff
@@ -1,41 +1,38 @@
...
 
 
-const SPEED = 300.0
-const JUMP_VELOCITY = -400.0
+@export var SPEED: int #you can guess
+@export var JUMP_VELOCITY = -400.0 #jump height kinda
+@export var rampin = 20 #how fast to go towards full speed
+@export var floaty: float #how far can jump
+@export var feather: int #how fast stop after space let go
+@export var friction: int #slow down kid
```
Comment (lishaduck) [inline wizard.gd:9]: This should probably use doc comments instead.

```suggestion
@export var SPEED: int ## You can guess.
@export var JUMP_VELOCITY = -400.0 ## Jump height, kinda.
@export var rampin = 20 ## How fast to go towards full speed.
@export var floaty: float ## How far the whizard can jump.
@export var feather: int ## How fast the whizard should stop after they stop jumping.
@export var friction: int ## Slow down, kid!
```

File: wizard.gd
Comment (lishaduck) [inline wizard.gd:1]: I'm not going to pretend to understand this, but it seems right.

File: wizard.gd [hunk 2]
```diff
@@ -1,41 +1,38 @@
...
 		velocity.y = JUMP_VELOCITY
+	if Input.is_action_just_released("jump") and not is_on_floor() and not velocity.y > 0:
+		velocity.y = move_toward(velocity.y,0,(feather+(abs(velocity.y/2))))
 
 	# Get the input direction and handle the movement/deceleration.
 	# As good practice, you should replace UI actions with custom gameplay actions.
 	var direction = Input.get_axis("left","right")
 	if direction:
-		velocity.x = direction * SPEED
+		velocity.x = move_toward(velocity.x,direction * SPEED,rampin+(abs(velocity.x))) #basicly takes rampin adds velocity over and over again
```
Comment (lishaduck) [inline wizard.gd:29]:
```suggestion
		velocity.x = move_toward(velocity.x, direction * SPEED, rampin + abs(velocity.x)) # This basically just takes rampin and adds velocity over and over again.
```

File: wizard.gd (hunk 1 above)
Comment (lishaduck) [reply wizard.gd:9]: Also, `rampin` should probably be `ramping`.

File: wizard.gd [hunk 3]
```diff
@@ -1,41 +1,38 @@
...
 
-# we can use look_at method for number tracking.
-
-
-
-
-
-#func _process(delta):
-#
-
```
Comment (lishaduck) [inline wizard.gd:41]: Thanks! Makes my heart feel happy 🤣

**Review by lishaduck — COMMENTED** [review]
Just a few comments, obviously nothing blocking a merge, seeing as you already self-merged it 😉.

**Review by iiAzui — COMMENTED** [review]
ok

File: wizard.gd (hunk 2 above)
Comment (iiAzui) [reply wizard.gd:29]: why would you delete the +

Comment (lishaduck) [reply wizard.gd:29]: Oh, that was a typo. Fixed!

**Pull request merged**